components/common/hooks/useGetLimits.ts

```diff
@@ -5,18 +5,25 @@ import {useState, useEffect, useMemo} from 'react';
 import {useDispatch, useSelector} from 'react-redux';
 
 import {Limits} from '@mattermost/types/cloud';
-import {getCloudLimits, getCloudLimitsLoaded, isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
+import {getCloudLimits, getCloudLimitsLoaded, isCurrentLicenseCloud, isCloudTrialExpired} from 'mattermost-redux/selectors/entities/cloud';
 import {getCloudLimits as getCloudLimitsAction} from 'actions/cloud';
 import {useIsLoggedIn} from 'components/global_header/hooks';
 
 export default function useGetLimits(): [Limits, boolean] {
     const isCloud = useSelector(isCurrentLicenseCloud);
+    const cloudTrialExpired = useSelector(isCloudTrialExpired);
     const isLoggedIn = useIsLoggedIn();
     const cloudLimits = useSelector(getCloudLimits);
     const cloudLimitsReceived = useSelector(getCloudLimitsLoaded);
     const dispatch = useDispatch();
     const [requestedLimits, setRequestedLimits] = useState(false);
 
+    useEffect(() => {
+        if (cloudTrialExpired) {
+            setRequestedLimits(false);
```

Comment:
@neallred when the cloud limits are changed, doesn't the webapp automatically update? Not sure if this would be needed?

When the trial expires it triggers a license and subscription change that should propagate all the way.